[system test] [doc] kafka package (#10650)
strimzi/strimzi-kafka-operator: development-docs/systemtests/io.strimzi.systemtest.kafka.KafkaST.md

| 2. | Create Kafka producer and consumer. | Kafka clients are successfully created. |
| 3. | Produce and consume messages using created clients. | Messages are successfully sent and received. |

**Labels:**

* [kafka](labels/kafka.md)

## testRegenerateCertExternalAddressChange

**Description:** Test regenerates certificates after changing Kafka's external address.

**Steps:**

| Step | Action | Result |
| - | - | - |
| 1. | Create Kafka without external listener | Kafka instance is created without an external listener |
| 2. | Edit Kafka to include an external listener | External listener is correctly added to the Kafka instance |
| 3. | Wait until the Kafka component has rolled | Kafka component rolls successfully with the new external listener |
| 4. | Compare Kafka broker secrets before and after adding external listener | Secrets are different before and after adding the external listener |

**Labels:**

* [kafka](labels/kafka.md)

## testRemoveComponentsFromEntityOperator

> **Description:** This test case verifies the correct deployment of Entity Operator, i.e., including both User Operator and Topic Operator. Entity Operator is firstly modified to exclude User Operator, afterwards it is modified to default configuration, which includes User Operator. The next step is removal of Topic Operator itself and finally, also removing User Operator, with Topic Operator being already removed.

Reviewer: ```suggestion
**Description:** This test case verifies the correct deployment of the Entity Operator, including both the User Operator and Topic Operator. First, the Entity Operator is modified to exclude the User Operator. Then, it's restored to its default configuration, which includes the User Operator. Next, the Topic Operator is removed, followed by the User Operator, with the Topic Operator already excluded.
```

**Steps:**

| Step | Action | Result |
| - | - | - |

> | 1. | Deploy Kafka with Entity Operator set. | Kafka is deployed, and Entity Operator consist of both Topic and User Operators |

Reviewer: ```suggestion
| 1. | Deploy Kafka with Entity Operator set. | Kafka is deployed, and Entity Operator consists of both Topic Operator and User Operator |
```
